R/string_operation.R

```diff
@@ -615,7 +615,58 @@ str_extract_inside <- function(column, begin = "(", end = ")") {
     stop("The end argument must be symbol such as ), }, ].")
   }
   exp = stringr::str_c("(?<=\\", begin, ").*?(?=\\", end, ")")
-  stringr::str_extract(column, exp)
+  if(all) {
+    stringr::str_extract_all(column, exp)
+  } else {
+    stringr::str_extract(column, exp)
+  }
+}
+
+#'Function to remove text inside the characters like bracket.
+#'@export
+str_remove_inside <- function(column, begin = "(", end = ")", all = FALSE){
+  if(stringr::str_length(begin) > 1) {
+    stop("The begin argument must be one character.")
+  }
+  if(stringr::str_length(end) > 1) {
+    stop("The end argument must be one character.")
+  }
+  if(grepl("[A-Za-z]", begin)) {
+    stop("The begin argument must be symbol such as (, {, [.")
+  }
+  if(grepl("[A-Za-z]", end)) {
+    stop("The end argument must be symbol such as ), }, ].")
+  }
+  exp = stringr::str_c("\\", begin, "[^()]*\\", end)
```

Comment:
This one fails to parse the following case. 

```
> str_remove_inside("abc[12(34)56]ijk", begin="[", end="]")
[1] "abc[12(34)56]ijk"
```

This should look like this.

```
  exp <- stringr::str_c("\\", begin, "[^\\" , begin, "\\", end, "]*\\", end)

```

Reply:
Also better using '<-' for assignment.

Reply:
Updated as suggested.